addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4750,8 +4749,8 @@ registry['sizing_x'] = registry.sizing.extend({
         // Below condition is added to remove offset of target element only
         // and not its children to avoid design alteration of a container/block.
         if (options.isCurrent) {
-            var _class = this.$target.attr('class').replace(/\s*(offset-xl-|offset-lg-)([0-9-]+)/g, '');
-            this.$target.attr('class', _class);
+            this.$target[0].className =
+                this.$target[0].className.replace(/\s*offset-(xl-|lg-)?([0-9-]+)/g, '');
```

Comment:
Since it's the same problem with `\b`, I'll make it bullet proof with `(?<!\S)` and close it with `(?!\S)`, and modify my other regex too.

Reply:
Why not `(\s+|^)<your_class_regex>(\s+|$)` maybe ? (not even sure what your regexes stand for ^^). Otherwise maybe avoiding the "complex" part of the regex is a good solution too, something like:
```javascript
classList.remove(...classList.filter(cls => cls.match(/offset-(xl-|lg-)?[0-9]+/)));
```
(something like that)
Honestly it maybe be more readable. But a solo regex is good too. But if you use `(?<!\S)`, make sure to comment what it is because it is not straightforward (to me at least).